model-engine/model_engine_server/inference/forwarding/http_forwarder.py

```diff
@@ -111,6 +118,47 @@ async def event_generator():
         return EventSourceResponse(event_generator())
 
 
+# This route is a catch-all for any requests that don't match the /predict or /stream routes
+# It will treat the request as a streaming request if the "stream" body parameter is set to true
+# NOTE: it is important for this to be defined AFTER the /predict and /stream endpoints
+# because FastAPI will match the first route that matches the request path
+async def predict_or_stream(
+    request: EndpointPredictV1Request,
+    background_tasks: BackgroundTasks,
+    sync_forwarder: Optional[Forwarder],
+    stream_forwarder: Optional[StreamingForwarder],
+    limiter=Depends(get_concurrency_limiter),
+):
+    if stream_forwarder and request.args and request.args.root.get("stream", False):
```

Comment:
if we don't like having this implicit expectation of having "stream" in the body, we can separate out a prefix for sync vs stream

e.g. 
/stream/{path} -> /{path} 
/sync/{path} -> /{path}

The tradeoff is that the gateway would have to know to prefix stream requests with /stream and sync with /sync. Both options have the potential to cause confusion in some way